src/jit/lsra.cpp

```diff
@@ -8536,6 +8545,13 @@ void LinearScan::recordMaxSpill()
         maxSpill[TYP_FLOAT] += 1;
     }
 #endif // _TARGET_X86_
+
+    if (compiler->opts.compUseSoftFP)
+    {
+        JITDUMP("Adding a spill temp for moving target address to a register.\n");
+        maxSpill[TYP_INT] += 1;
```

Comment:
@CarolEidt After I made the second commit(I thought it would be enough), but I got the assertion mentioned from https://github.com/dotnet/coreclr/pull/13023#issuecomment-319254128. So I had to fix it, and the code here is sort of a workaround.